text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,401 @@
+
+Team member is a mix of RFC 2689's "working group participant" and
+"compiler team contributor" roles. It is explicitly intended to be granted more
+liberally to contributors who have demonstrated competence and trustworthiness,
+for whom they would be able to work more effectively with these permissions and
+can be trusted to use them responsibly. Team members do not need to
+have experience with most of the compiler, and can be specialised to specific
+subsystems of the compiler.
+
+Team members are granted *r+*, *try*, *triagebot*, *rustc-perf*, and
+*crater* permissions; *organization membership*; and *dev desktop* access.
+Team members are considered members of the Rust project as a whole, and
+are automatically eligible for any benefits that incurs (e.g. invitations to
+meetups of project members). As representatives of the Rust project, trusted
+contributors are expected to obey not just the letter of the
+[Code of Conduct][coc] but its spirit.
+
+Team members can choose to take on additional responsibilities, such as
+those listed in the [responsibilities][responsibilities] section. Participating
+in the team's review rotation is encouraged.
+
+If a team member becomes inactive (the contributor's prior contributions and
+other interactions with the project cease) for longer than a year, the team
+member will be moved into alumni status. At any point in the future, they can
+ask to be re-instated at the team member level if they desire.
+
+[Compiler Development Guide]: https://rustc-dev-guide.rust-lang.org/
+
+Maintainers
+-----------
+[maintainers]: #maintainers
+
+Team members are eligible to become maintainers after they have continued to
+contribute actively for a year. Team members can contact team leads or will
+be contacted by team leads to enquire about promotion to maintainership. Team
+members who are eligible for maintainership do not have to become maintainers.
+
+Unlike team members, maintainers are expected to consider themselves
+as exactly that, *maintainers*, of the compiler - put otherwise, to be invested
+in the quality of the compiler codebase and overall health of the compiler
+team, independent of their own projects. Maintainership is primarily intended
+to recognise and encourage participation in activities which are vital to the
+success of the compiler team and broader project.
+
+Maintainers are expected to participate in the ongoing maintenance tasks
+that the compiler team is responsible for (with all of the expected caveats
+for vacation time, mental health breaks, etc) - listed as
+[responsibilities][responsibilities] below. However, not all maintainers need
+to participate in these responsibilities to an equal degree. Maintainers should
+participate in these tasks to the degree that they are able - volunteers are not
+expected to participate as much as those employed to work on the compiler, for
+example. It is the responsibility of the compiler team leads to ensure that the
```

Comment:
I think there is an unstated consequence here that if maintainers are not actively participating in compiler team maintenance (minus stated caveats), they can be removed as maintainers and this is as the discretion of team leads.

Reply:
Added a little bit of wording for this, may want more, let me know what you think.